lib/captcha.ts

```diff
@@ -0,0 +1,91 @@
+import prisma from "./prisma";
+
+const generateRandomNumberInRange = (start: number, end: number) =>
+  Math.floor(Math.random() * end) + start;
+
+/**
+ * Gets a captcha challenge record from the database
+ * @param id - The id of the captcha challenge record in the database
+ * @returns The captcha challenge record from the database
+ **/
+export const getCaptchaChallenge = async (id: number) => {
+  return await prisma.captchaChallenge.findUnique({
+    where: {
+      id: id,
+    },
+  });
+};
+
+/**
+ * Generates a new captcha challenge record in the database
+ * @param userAddress - The address of the user that is trying to solve the captcha
+ * @param waitlistId - The id of the waitlist that the user is trying to join
+ * @returns The id of the new captcha challenge record created in the database and the two numbers the user needs to add
+ **/
+export const generateCaptchaChallenge = async (
+  userAddress: string,
+  waitlistId: number
+) => {
+  const numA = generateRandomNumberInRange(10, 20);
+  const numB = generateRandomNumberInRange(10, 20);
+  const newChallenge = await prisma.captchaChallenge.create({
+    data: {
+      waitlistId: waitlistId,
+      userAddress: userAddress,
```

Comment:
I think this should be FID, as the user interacting has a unique FID, while address can be shared among different FIDs